### PR TITLE
PWGHF: Adding Bayesian PID selections in the Lc->pKpi selector task

### PR DESCRIPTION
**Pull request by DelloStritto**
In this pull request you can find:

 1. Implementation of the bayesian selection in the Lc->pKpi selector
 2. Implementation of some bayesian methods in the trackSelectorPID class.
 3. Histogram addition in the Lc->pKpi task

@vkucera

### CODE REVIEW

File: PWGHF/TableProducer/HFLcCandidateSelector.cxx [hunk 1]
```diff
@@ -45,6 +45,10 @@ struct HFLcCandidateSelector {
   Configurable<double> d_pidTOFMaxpT{"d_pidTOFMaxpT", 2.5, "Upper bound of track pT for TOF PID"};
   Configurable<double> d_nSigmaTOF{"d_nSigmaTOF", 3., "Nsigma cut on TOF only"};
   Configurable<double> d_nSigmaTOFCombined{"d_nSigmaTOFCombined", 5., "Nsigma cut on TOF combined with TPC"};
+  // BAYES
```
Comment (vkucera) [inline PWGHF/TableProducer/HFLcCandidateSelector.cxx:48]:
```suggestion
  // Bayesian
```

File: Common/Core/TrackSelectorPID.h [hunk 2]
```diff
@@ -46,7 +46,36 @@ class TrackSelectorPID
     PIDAccepted
   };
 
-  void setPDG(int pdg) { mPdg = std::abs(pdg); }
+  void setPDG(int pdg)
```
Comment (vkucera) [inline Common/Core/TrackSelectorPID.h:49]: To make sure that both `mPdg` and `mSpecies` are set in the standard constructor, `setPDG(pdg)` should replace `: mPdg(std::abs(pdg))` there.

File: PWGHF/TableProducer/HFLcCandidateSelector.cxx [hunk 3]
```diff
@@ -45,6 +45,10 @@ struct HFLcCandidateSelector {
...
+  Configurable<bool> d_BayesPID{"d_BayesPID", true, "Bool to use or not the PID based on bayesian probability cut at filtering level"};
```
Comment (vkucera) [inline PWGHF/TableProducer/HFLcCandidateSelector.cxx:49]:
```suggestion
  Configurable<bool> d_BayesPID{"d_BayesPID", true, "Bool to use or not the PID based on Bayesian probability cut at filtering level"};
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
@iarsene Can you please merge this?

**Review by vkucera — APPROVED** [review]

**Review by ktf — APPROVED** [review]

**Pull request merged**